includes/functions/functions_email.php

```diff
@@ -359,7 +359,21 @@ function zen_mail($to_name, $to_address, $email_subject, $email_text, $from_emai
       /**
        * Send the email. If an error occurs, trap it and display it in the messageStack
        */
-      if (!$mail->send()) {
+      try { 
+         if (!$mail->send()) {
+           $msg = sprintf(EMAIL_SEND_FAILED . '&nbsp;'. $mail->ErrorInfo, $to_name, $to_email_address, $email_subject);
+           if ($messageStack !== NULL) {
+             if (IS_ADMIN_FLAG === true) {
+               $messageStack->add_session($msg, 'error');
+             } else {
+               $messageStack->add('header', $msg, 'error');
+             }
+           } else {
+             error_log($msg);
+           }
+           $ErrorInfo .= ($mail->ErrorInfo != '') ? $mail->ErrorInfo . "\n" : '';
+         }
```

Comment:
Is all this duplicated code needed? If it's throwing an exception when it fails to send, then won't it immediately fall back to the `catch` block, making this duplicated code redundant?

Reply:
Edit: It looks like it still returns a boolean, which we will also need to check.

Can we simplify this so there's no copy/paste of the same logic? Perhaps just put the $mail->send() in the try block, and leave all the failure-checking after the closed (empty) catch block?

Reply:
I know there's duplication - just wanted to start by getting the concept out there.  We could try your idea.